src/kafka-util/src/client.rs

```diff
@@ -239,18 +244,36 @@ pub struct BrokerRewrite {
 #[derive(Clone)]
 pub struct BrokerRewritingClientContext<C> {
     inner: C,
-    rewrites: BTreeMap<BrokerAddr, Arc<dyn Fn() -> BrokerRewrite + Send + Sync>>,
+    rewrites: Arc<Mutex<BTreeMap<BrokerAddr, Arc<dyn Fn() -> BrokerRewrite + Send + Sync>>>>,
```

Comment:
Should this be a non-async `Mutex`? It's never held across an `await` point.

Reply:
its to avoid blocking a tokio task when calling `add_broker_rewrite`, which holds the lock across blocking io

Reply:
Wait, sorry, where does it do that? All calls to `add_broker_rewrite` that I can see are super simple closures that basically just construct a `BrokerRewrite` out of closed-over state.

Reply:
We still need to store an arbitrary ssh token

hmm, i think you are right, this might be able to be a sync lock, the `SshTunnelManager` prevents races creating too many ssh connections at once

Reply:
i think itll be more like:
```
enum BrokerRewriteHandle {
    Simple(BrokerRewrite),
    SshTunnel(BrokerRewrite, ManagedSshTunnelHandle),
}
```

but yeah, seems reasonable!